cmds/insmod/insmod.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"syscall"
+	"unsafe"
+)
+
+func main() {
+	var options string
+
+	if len(os.Args) < 2 {
+		log.Fatalf("insmod: ERROR: missing filename.\n")
+	}
+
+	// get filename from argv[1]
+	filename := os.Args[1]
+
+	// Everything else is module options
+	for i := 2; i < len(os.Args); i++ {
+		options = strings.Join([]string{options, os.Args[i]}, " ")
+	}
+
+	// read file into memory
```

Comment:
Ah, I was doing my thing where I outline each step in comments and then fill in the code.